File: app/src/main/java/com/example/githubuserssample/networking/ApiService.java
package com.example.githubuserssample.networking;

import com.example.githubuserssample.models.UserDetails;
import com.example.githubuserssample.models.UsersList;

import okhttp3.ResponseBody;
import retrofit2.Call;
import retrofit2.http.Field;
import retrofit2.http.GET;
import retrofit2.http.Path;
import retrofit2.http.Query;

public interface ApiService {
    //define your endpoints here

    @GET("users/{username}")
    Call<UserDetails>  getUserDetails(@Path("username") String username);

    @GET("search/users")
    Call<UsersList> getUsersByLocation(
            @Query("q") String location);
}

File: app/src/main/java/com/example/githubuserssample/networking/ApiService.kt
package com.example.githubuserssample.networking

import com.example.githubuserssample.models.UserDetails
import com.example.githubuserssample.models.UsersList
import retrofit2.http.GET
import retrofit2.http.Path
import retrofit2.http.Query

interface ApiService {
    //define your endpoints here
    @GET("users/{username}")
    suspend fun getUserDetails(@Path("username") username: String?): UserDetails

    @GET("search/users")
     suspend fun getUsersByLocation(@Query("q") location: String?): UsersList
}